data/World/Ganons Castle.json

```diff
@@ -97,7 +126,12 @@
                 has_bombchus and (logic_lens_castle or can_use(Lens_of_Truth))",
             "Nut Pot": "
                 (logic_spirit_trial_hookshot or Progressive_Hookshot) and
-                has_bombchus and Bow and Mirror_Shield"
+                has_bombchus and Bow and Mirror_Shield",
+            "Ganons Castle Spirit Trial Recovery Heart": "True",
+            "Ganons Castle Spirit Trial Pot 1" : "can_use(Mirror_Shield) and has_bombchus and
+                (logic_spirit_trial_hookshot or Progressive_Hookshot)",
+            "Ganons Castle Spirit Trial Pot 2" : "can_use(Mirror_Shield) and has_bombchus and
+                (logic_spirit_trial_hookshot or Progressive_Hookshot)"
```

Comment:
For the pots at the end of Spirit Trial, you need to check for bow to burn the web.